apps/glusterfs/volume_entry.go

```diff
@@ -154,6 +155,34 @@ func NewVolumeEntryFromId(tx *bolt.Tx, id string) (*VolumeEntry, error) {
 	return entry, nil
 }
 
+func NewVolumeEntryFromClone(v *VolumeEntry, name string) *VolumeEntry {
+	entry := NewVolumeEntry()
+
+	entry.Info.Id = utils.GenUUID()
+	if name == "" {
+		entry.Info.Name = "vol_" + entry.Info.Id
+	} else {
+		entry.Info.Name = name
+	}
+
+	entry.GlusterVolumeOptions = v.GlusterVolumeOptions
+	entry.Info.Cluster = v.Info.Cluster
+	entry.Info.Durability = v.Info.Durability
+	entry.Info.Durability.Type = v.Info.Durability.Type
+	entry.Info.Gid = v.Info.Gid
+	entry.Info.Mount = v.Info.Mount
+	entry.Info.Size = v.Info.Size
+	entry.Info.Snapshot = v.Info.Snapshot
+	copy(entry.Info.Mount.GlusterFS.Hosts, v.Info.Mount.GlusterFS.Hosts)
+	entry.Info.Mount.GlusterFS.MountPoint = v.Info.Mount.GlusterFS.Hosts[0] + ":" + entry.Info.Name
+	entry.Info.Mount.GlusterFS.Options = v.Info.Mount.GlusterFS.Options
+	entry.Info.BlockInfo.FreeSize = v.Info.BlockInfo.FreeSize
+	copy(entry.Info.BlockInfo.BlockVolumes, v.Info.BlockInfo.BlockVolumes)
```

Comment:
I am very unsure about this. Should we even allow for clone of blockhosting volumes? What happens to the IQN of the blocks that reside on them. @pkalever  How unique is the IQN? If blockhosting volume is cloned along with the metadata and the block files on it, would it be fine from gluster-block perspective?

Reply:
Caveat: don't confuse block volumes and block-hosting volumes (glusterfs volumes that host block volumes)

Reply:
> Caveat: don't confuse block volumes and block-hosting volumes (glusterfs volumes that host block volumes)

Good point, but can you currently do anything useful with block hosting volumes that have block volumes in them with a clone? (And cloning an empty volume is pretty pointless) My first impression is no, because none of th iscsi stuff will be configured so the clone would just exist  but not be usable. 

> Maybe a HTTP code like EINVAL exists?

You're probably stuck with 400 as I don't see any of the other codes fitting this well. But please emit error text that is specific. Something like "cloning block hosting volumes is not supported" for example

Reply:
@nixpanic, I implemented a check for Info.Block in prepareVolumeClone() in an additional commit. Please review.